src/odemis/gui/cont/acquisition/overview_stream_acq.py

```diff
@@ -103,3 +103,60 @@ def open_acquisition_dialog(self):
             return acq_dialog.data
         else:
             return None
+
+
+class CorrelationDialogController(object):
+    """Controller to handle the multipoint correlation (3DCT) initialization"""
+
+    def __init__(self, tab_data, tab):
+        """
+        tab_data (MicroscopyGUIData): the representation of the microscope GUI
+        tab: (Tab): the tab which should show the data
+        """
+        self._tab_data_model = tab_data
+        self._main_data_model = tab_data.main
+        self._tab = tab
+
+    def open_correlation_dialog(self):
+        """
+        Opens the multipoint correlation dialog
+        """
+        # save the original settings
+        settingsbar_controller = self._tab.settingsbar_controller
+        orig_entries = get_global_settings_entries(settingsbar_controller)
+        for sc in self._tab.streambar_controller.stream_controllers:
+            orig_entries += get_local_settings_entries(sc)
+        orig_settings = preset_as_is(orig_entries)
+        settingsbar_controller.pause()
+        settingsbar_controller.enable(False)
+
+        # pause all the live acquisitions
+        streambar_controller = self._tab.streambar_controller
+        streambar_controller.pauseStreams()
+        streambar_controller.pause()
+        streambar_controller.enable(False)
+
+        # create the dialog
+        try:
+            cor_dialog = CorrelationDialog(
+                self._tab.main_frame, self._tab_data_model)
+            parent_size = [int(v * 0.9) for v in self._tab.main_frame.GetSize()]
+
+            cor_dialog.SetSize(parent_size)
+            cor_dialog.Center()
+            cor_dialog.ShowModal()
+
+        except Exception:
+            logging.exception("Failed to create correlation dialog")
+            raise
+        finally:
+            apply_preset(orig_settings)
+
+            settingsbar_controller.enable(True)
+            settingsbar_controller.resume()
+
+            streambar_controller.enable(True)
+            streambar_controller.resume()
+
+            cor_dialog.correlation_points_controller.stop()
```

Comment:
You can just override the `CorrelationDialog.Destroy` method to call `correlation_points_controller.stop()` when it's being destroyed. This way, the callers don't need to care about such details.